chalicelib/responseobjects/hca_response_v5.py

```diff
@@ -122,26 +125,52 @@ class ManifestResponse(AbstractResponse):
     """
     Class for the Manifest response. Based on the AbstractionResponse class
     """
+
     def return_response(self):
-        return self.apiResponse
+        es_search = self.es_search
+
+        def field_mapper():
+            for hit in es_search.scan():
+                hit_dict = hit.to_dict()
+                mapped_dict = {}
+                for friendly_name_key, es_name_value in self.mapping.iteritems():
+                    es_val = jmespath.search(es_name_value, hit_dict)
+                    if es_val and friendly_name_key in self.manifest_entries:
+                        mapped_dict[friendly_name_key] = es_val
+                yield mapped_dict
+
+        headers = {'Content-Disposition': 'attachment; filename="export.tsv"',
+                   'Content-Type': 'text/tab-separated-values'}
+
+        output = BytesIO()
+        mapped_dict = field_mapper().next()
```

Comment:
This seems to start one iteration and then make one expensive round-trip to ES but then abandons the iteration to start another one in the loop below.

Reply:
I think `.next()` does not exist on generators. At least not in Python3. You'd have to use the `next()` builtin function.

Has this code ever been tested?